addons/mail/static/src/core/common/message_model.js

```diff
@@ -65,6 +65,9 @@ export class Message extends Record {
             return otherDidNotSee.length === 0;
         },
     });
+    threadAsFirstUnreadMessage = Record.one("Thread", {
+        inverse: "firstUnreadMessage",
+    });
```

Comment:
Note that `Message` suffix could be removed from this field, as this is redundant by the model name.
This should be kept for the inverse field `firstUnreadMessage` though